app/src/main/java/net/squanchy/navigation/HomeActivity.java

```diff
@@ -104,8 +110,32 @@ protected void onStart() {
         }
     }
 
+    private void askProximityPersmissionToStartRadar() {
+        int permissionCheck = ContextCompat.checkSelfPermission(this,
+                Manifest.permission.ACCESS_FINE_LOCATION);
+        if (permissionCheck == PackageManager.PERMISSION_GRANTED) {
+            proximityService.startRadar();
+        } else {
+            ActivityCompat.requestPermissions(this,
+                    new String[]{Manifest.permission.ACCESS_FINE_LOCATION},
+                    MY_PERMISSIONS_REQUEST_ACCESS_FINE_LOCATION);
+        }
+    }
+
+    @Override
+    public void onRequestPermissionsResult(int requestCode, @NonNull String[] permissions, @NonNull int[] grantResults) {
+        if (requestCode == MY_PERMISSIONS_REQUEST_ACCESS_FINE_LOCATION)
+            if (grantResults.length > 0
```

Comment:
Extract a method for the check for readability, e.g., `hasGrantedFineLocationAccess()`